### PR TITLE
Ability to decrypt kms encrypted values

### PR DESCRIPTION
**Pull request by achinthagunasekara**
#### SUMMARY

Adding ability to decrypt values encrypted with AWS KMS. (and encrypt values with AWS KMS)

in `my_role/defaults/main.yaml` value can be included that is encrypted with AWS KMS.

```
encrypted_value: >-
  {{ abcd123abcd13 abcd123abcd13abcd12abcd13
  abcd123abcd13abcd123abcd13abcd123ab1d1312
  abcd123abcd13abcd123abcd13abcd123ssdsas12
  sadsdsasd | kms_decrypt(KEY_ARN) }}
```

### CODE REVIEW

File: lib/ansible/plugins/filter/aws_kms.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
+""" Modoule to handle encrypting and decrypting of items with KMS """
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type  # pylint: disable=invalid-name
+import base64
+import botocore
```
Comment (mattclay) [inline lib/ansible/plugins/filter/aws_kms.py]: You'll need a try/catch around imports that are not from the Python standard library. See an example here:

https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/filter/json_query.py#L26

Comment (achinthagunasekara) [reply lib/ansible/plugins/filter/aws_kms.py]: Thanks @mattclay I've updated my changes accordingly :)

File: lib/ansible/plugins/filter/aws_kms.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    import botocore
+    import aws_encryption_sdk
+    HAS_DEPENDENCIES = True
+except ImportError:
+    HAS_DEPENDENCIES = False
+
+
+def aws_kms_encrypt(plaintext, key_arn):
+    """ Encrypt with KMS """
+    if not HAS_DEPENDENCIES:
```
Comment (jitran) [inline lib/ansible/plugins/filter/aws_kms.py]: Hi Archie, is it worth moving this conditional block into an internal function as it's repeated twice?

Comment (achinthagunasekara) [reply lib/ansible/plugins/filter/aws_kms.py]: @jitran Good point. Updated with proposed changes.

**Comment by achinthagunasekara** [comment]
Closing this PR in favour of https://github.com/ansible/ansible/pull/40575

**Comment by achinthagunasekara** [comment]
Closing this PR in favour of https://github.com/ansible/ansible/pull/40575

**Pull request closed without merging**